### DISCUSSION TITLE
Any abstraction like LangGraph?

### DISCUSSION THREAD

**Discussion by npuichigo** [0xPlaygrounds/rig#167] [General]
It's really a fantastic project. And I have been looking for such a project written in Rust to replace my experience with LangChain. One question is that LangGraph seems to be a supplement for LangChain, and maybe there's something that's missing in using LangChain only. So I would like to ask do we have any further plan or more powerful agent abstraction to be an alternative to LangGraph in the future?

**Comment by 0xMochan** [comment]
In rig 0.6, a new feature called agent pipelines allows you to organize data flow between agents together. This is at its primitive atm but we'll be expanding this in future releases.

**Reply by npuichigo** [reply]
That's cool. Express DAG in native async Rust is very intuitive.
